### PR TITLE
fix(curriculum): changed test text to use should for Coding Interview Prep - part 2 of 2

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Related to PR #36860 

This PR makes sure all challenges' tests in the following sections of the `Coding Interview Prep` section use "should " or "should not".

* Project Euler
* Rosetta Code

### CODE REVIEW

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/date-format.english.md [hunk 1]
```diff
@@ -25,13 +25,13 @@ Example output: <code>['2007-11-23', 'Sunday, November 23, 2007']</code>
...
 ```yml
 tests:
-  - text: <code>getDateFormats</code> is a function.
+  - text: <code>getDateFormats</code> should be a function.
     testString: assert(typeof getDateFormats === 'function');
-  - text: Should return an object.
+  - text: <code>getDateFormats</code> should return an object.
     testString: assert(typeof getDateFormats() === 'object');
-  - text: Should returned an array with 2 elements.
+  - text: <code>getDateFormats</code> should returned an array with 2 elements.
```
Comment (moT01) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/date-format.english.md]:
```suggestion
  - text: <code>getDateFormats</code> should return an array with 2 elements.
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/topological-sort.english.md [hunk 2]
```diff
@@ -57,15 +57,15 @@ There are two popular algorithms for topological sorting:
 
 ```yml
 tests:
-  - text: <code>topologicalSort</code> is a function.
+  - text: <code>topologicalSort</code> should be a function.
     testString: assert(typeof topologicalSort === 'function');
-  - text: <code>topologicalSort</code> must return correct library order..
+  - text: <code>topologicalSort</code> should return correct library order..
```
Comment (moT01) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/topological-sort.english.md]:
```suggestion
  - text: <code>topologicalSort</code> should return correct library order.
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/topological-sort.english.md [hunk 3]
```diff
@@ -57,15 +57,15 @@ There are two popular algorithms for topological sorting:
...
     testString: assert.deepEqual(topologicalSort(libsSimple), ['bbb', 'aaa']);
-  - text: <code>topologicalSort</code> must return correct library order..
+  - text: <code>topologicalSort</code> should return correct library order..
```
Comment (moT01) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/topological-sort.english.md]:
```suggestion
  - text: <code>topologicalSort</code> should return correct library order.
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/topological-sort.english.md [hunk 4]
```diff
@@ -57,15 +57,15 @@ There are two popular algorithms for topological sorting:
...
     testString: assert.deepEqual(topologicalSort(libsVHDL), solutionVHDL);
-  - text: <code>topologicalSort</code> must return correct library order..
+  - text: <code>topologicalSort</code> should return correct library order..
```
Comment (moT01) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/topological-sort.english.md]:
```suggestion
  - text: <code>topologicalSort</code> should return correct library order.
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/topological-sort.english.md [hunk 5]
```diff
@@ -57,15 +57,15 @@ There are two popular algorithms for topological sorting:
...
     testString: assert.deepEqual(topologicalSort(libsCustom), solutionCustom);
-  - text: <code>topologicalSort</code> must ignore unorderable dependencies..
+  - text: <code>topologicalSort</code> should ignore unorderable dependencies..
```
Comment (moT01) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/topological-sort.english.md]:
```suggestion
  - text: <code>topologicalSort</code> should ignore unorderable dependencies.
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/y-combinator.english.md [hunk 6]
```diff
@@ -27,17 +27,17 @@ Define the stateless Y combinator function and use it to compute <a href="https:
 
 ```yml
 tests:
-  - text: Y must return a function
+  - text: Y mushouldst return a function.
```
Comment (moT01) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/y-combinator.english.md]:
```suggestion
  - text: Y should return a function.
```

**Review by moT01 — CHANGES_REQUESTED** [review]
Couple minor things...

**Comment by RandellDawson** [comment]
@moT01 I committed your suggested changes.  Thanks for the review.

**Review by moT01 — APPROVED** [review]
🎉 LGTM @RandellDawson 🎉

**Review by Manish-Giri — APPROVED** [review]
LGTM 💯

**Pull request merged**